R/aes.r

```diff
@@ -334,3 +334,51 @@ mapped_aesthetics <- function(x) {
   is_null <- vapply(x, is.null, logical(1))
   names(x)[!is_null]
 }
+
+
+#' Check a mapping for discouraged usage
+#'
+#' Checks that `$` and `[[` are not used when the target *is* the data
+#'
+#' @param mapping A mapping created with [aes()]
+#' @param data The data to be mapped from
+#'
+#' @noRd
+check_aes_extract_usage <- function(mapping, data) {
+  lapply(mapping, check_aes_extract_usage_quo, data)
+}
+
+check_aes_extract_usage_quo <- function(quosure, data) {
```

Comment:
I think you could eliminate this function now (and inline into `check_aes_extract_usage()`)